### DISCUSSION TITLE
Pod is terminated but DAG is still in running state

### DISCUSSION THREAD

**Discussion by shihabcsedu09** [apache/airflow#14315] [Q&A] [answered]
I have configured AKS Kubernetes Cluster where there is an Airflow Scheduler pod and an Airflow Scheduler pod. I have written a DAG that will run using the Kubernetes Pod Operator.

When the DAG is triggered a pod is created and the steps inside the DAG start running. The issue arises when the task is done and the pod moves to a terminated/completed state the airflow webserver seems to not receive that information. For this reason, the DAG status is always running in the webserver.

I observed both the Kubernetes pod logs and the logs that we can see from the web UI for that task. The logs in the webserver are lagging behind from the logs in the pod.

I am using Airflow 1.10.14.

**Comment by shihabcsedu09** [comment]
Its also happening in Airflow 2.0.1. What I observed is that it happens when there is a long-running airflow step.

**Comment by turbaszek** [comment, marked as answer]
Please open a bug report. It would be best if you could provide a DAG which can replicate the issue. From what you told it should be simple sleep task.

There are some other issues related to state updated when using k8s operators:
https://github.com/apache/airflow/issues/10325
https://github.com/apache/airflow/issues/11190

**Reply by shihabcsedu09** [reply]
@turbaszek I have created an issue - [14791](https://github.com/apache/airflow/issues/14791)
